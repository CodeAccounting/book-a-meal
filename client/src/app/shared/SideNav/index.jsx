import React from 'react';
import PropTypes from 'prop-types';
import LinkBtn from '../../shared/Link';
import { userPropType } from '../../../helpers/proptypes';

/**
 * @exports
 * @function SideNav
 * @returns {JSX} SideNav
 */
const SideNav = ({ user, logout }) => (
  <div className="sidenav" id="sidenav">
    <div className="sidenav-content">
      <div className="sidenav-header">
        <div className="close d-none-md">
          <a href="#">&times;</a> {/* eslint-disable-line */}
        </div>
        <div className="sidenav-title">
          <h3>
            <a href="/">BOOK-A-MEAL</a>
          </h3>
          <div className="username-circle">
            <p>{user.businessName.substring(0, 1)}</p>
          </div>
          <p>{user.businessName}</p>
          <span>
            <LinkBtn clickHandler={logout}>Logout</LinkBtn>
          </span>
        </div>
      </div>
      <div className="sidenav-body">
        <a href="/" className="menu-item active">Dashboard</a>
        <a href="/" className="menu-item">Meals</a>
        <a href="/" className="menu-item">Menu</a>
        <a href="/" className="menu-item">Order History</a>
      </div>
    </div>
  </div>
);

SideNav.propTypes = {
  ...userPropType,
  logout: PropTypes.func.isRequired
};

export default SideNav;
