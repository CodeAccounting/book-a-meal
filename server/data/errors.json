{
  "401": "Unauthorized",
  "403": "Forbidden",
  "404": "Not Found",
  "422": "Validation Error",
  "500": "Internal Server Error",
  "empty": "Empty PUT Requests Are Not Allowed"
}