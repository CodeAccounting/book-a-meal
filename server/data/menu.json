[
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "date": "2018-05-06",
     "meals": [
        "81211c24-51c0-46ec-b1e0-18db55880958",
        "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
        "baa0412a-d167-4d2b-b1d8-404cb8f02631",
        "f9eb7652-125a-4bcb-ad81-02f84901cdc3"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "1adfcfe7-c66d-42d2-82fd-39c1decd290a",
     "date": "2018-04-06",
     "meals": [
        "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
        "3ec802c6-0a32-4d29-b27b-42e0f4b532dd"
     ],
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
     "menuId": "3a4fb034-6e52-4bde-9abc-91119a39478d",
     "date": "2018-05-07",
     "meals": [
        "72a3417e-45c8-4559-8b74-8b5a61be8614"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
     "menuId": "f43f3d49-a6c9-476d-b65a-1af772ff0f36",
     "date": "2018-04-06",
     "meals": [
        "72a3417e-45c8-4559-8b74-8b5a61be8614",
        "46ced7aa-eed5-4462-b2c0-153f31589bdd"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
     "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
     "date": "2018-04-27",
     "meals": [
        "91b6e41c-0972-4ac5-86da-4ac1f5226e83",
        "f9eb7652-125a-4bcb-ad81-02f84901cdc3",
        "8a65538d-f862-420e-bcdc-80743df06578"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  }
]