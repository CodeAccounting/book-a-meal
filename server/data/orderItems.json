[
  {
     "id": 1,
     "mealId": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "quantity": 2,
     "createdAt": "2018-04-06T14:34:90.000",
     "updatedAt": "2018-04-06T14:34:90.000"
  },
  {
     "id": 2,
     "mealId": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "quantity": 1,
     "createdAt": "2018-04-06T14:34:90.000",
     "updatedAt": "2018-04-06T14:34:90.000"
  }, 
  {
    "id": 3,
    "mealId": "72a3417e-45c8-4559-8b74-8b5a61be8614",
    "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
    "quantity": 2,
    "createdAt": "2018-04-06T14:34:90.000",
    "updatedAt": "2018-04-06T14:34:90.000"
  },
  {
    "id": 4,
    "mealId": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
    "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
    "quantity": 2,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000"
  },
  {
    "id": 5,
    "mealId": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
    "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
    "quantity": 2,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000"
  }
]