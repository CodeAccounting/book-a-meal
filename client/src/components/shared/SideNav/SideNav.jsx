import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import LinkBtn from '../Link';
import CloseIcon from '../CloseIcon';
import { userPropTypes } from '../../../helpers/proptypes';
import './SideNav.scss';

/**
 * @exports
 * @function SideNav
 * @returns {JSX} SideNav
 */
class SideNav extends Component {
  static propTypes = {
    ...userPropTypes,
    logout: PropTypes.func.isRequired,
    active: PropTypes.string.isRequired
  };

  /**
   * @memberof SideNav
   * @returns {nothing} nothing
   */
  handleToggleSideNavClick = () => this.props.toggleSideNav()

  /**
   * @memberof SideNav
   * @returns {JSX} NavLinks
  */
  renderNavLinks = () => {
    const { active } = this.props;

    return (
      <div className="sidenav-body">
        <Link href="/" to="/" className={`menu-item ${active === 'dashboard' && 'active'}`}>Dashboard</Link>
        <Link href="/meals" to="/meals" className={`menu-item ${active === 'meals' && 'active'}`}>Meals</Link>
        <Link href="/menu" to="/menu" className={`menu-item ${active === 'menu' && 'active'}`}>Menu</Link>
        <Link href="/orders" to="/orders" className={`menu-item ${active === 'orders' && 'active'}`}>Orders</Link>
      </div>
    );
  }

  /**
   * @memberof SideNav
   * @returns {JSX} SideNav Component
  */
  render() {
    const { user, logout, open } = this.props;

    return (
      <div className={`sidenav ${open ? 'show' : null}`} id="sidenav">
        <div className="sidenav-content">
          <div className="sidenav-header">
            <CloseIcon divClass="d-none-md" clickHandler={this.handleToggleSideNavClick} />
            <div className="sidenav-title">
              <h3><Link href="/" to="/">BOOK-A-MEAL</Link></h3>
              <div className="username-circle"><p>{user.businessName.substring(0, 1)}</p></div>
              <p>{user.businessName}</p>
              <span><LinkBtn clickHandler={logout}>Logout</LinkBtn></span>
            </div>
          </div>
          {this.renderNavLinks()}
        </div>
      </div>
    );
  }
}

export default SideNav;
