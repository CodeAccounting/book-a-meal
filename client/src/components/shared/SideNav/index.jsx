import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { push } from 'react-router-redux';
import SideNav from './SideNav';
import { toggleSideNav } from '../../../store/actions/ui';

const mapStateToProps = state => ({
  open: state.ui.sideNav.open
});

const mapDispatchToProps = dispatch => bindActionCreators({
  toggleSideNav, push
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(SideNav);
