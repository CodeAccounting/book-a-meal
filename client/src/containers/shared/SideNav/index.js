import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { push } from 'react-router-redux';
import SideNav from '../../../components/shared/SideNav';
import { toggleSideNav } from '../../../actions/ui';

const mapStateToProps = state => ({
  open: state.ui.sideNav.open
});

const mapDispatchToProps = dispatch => bindActionCreators({
  toggleSideNav, push
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(SideNav);
