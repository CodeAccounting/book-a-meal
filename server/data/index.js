import users from './users.json';
import meals from './meals.json';
import menu from './menu.json';
import orders from './orders.json';
import errors from './errors.json';

export default {
  users, meals, menu, orders, errors
};
