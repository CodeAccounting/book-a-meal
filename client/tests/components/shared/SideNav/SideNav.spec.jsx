import React from 'react';
import SideNav from '../../../../src/app/shared/SideNav';
import { caterer } from '../../../setup/data';

describe('SideNav', () => {
  it('renders correctly', () => {
    const shallowWrapper = shallow(<SideNav user={caterer} logout={jest.fn()} active="dashboard" />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('renders correctly when meals tab is active', () => {
    const shallowWrapper = shallow(<SideNav user={caterer} logout={jest.fn()} active="meals" />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('renders correctly when menu tab is active', () => {
    const shallowWrapper = shallow(<SideNav user={caterer} logout={jest.fn()} active="menu" />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('renders correctly when orders tab is active', () => {
    const shallowWrapper = shallow(<SideNav user={caterer} logout={jest.fn()} active="orders" />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});
