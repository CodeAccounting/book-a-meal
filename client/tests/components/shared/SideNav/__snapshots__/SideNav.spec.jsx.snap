// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav renders correctly 1`] = `
<div
  className="sidenav null"
  id="sidenav"
>
  <div
    className="sidenav-content"
  >
    <div
      className="sidenav-header"
    >
      <CloseIcon
        btnID={null}
        clickHandler={[Function]}
        divClass="d-none-md"
      />
      <div
        className="sidenav-title"
      >
        <h3>
          <Link
            href="/"
            replace={false}
            to="/"
          >
            BOOK-A-MEAL
          </Link>
        </h3>
        <div
          className="username-circle"
        >
          <p>
            F
          </p>
        </div>
        <p>
          Food Circle
        </p>
        <span>
          <Link
            className={null}
            clickHandler={[MockFunction]}
            id={null}
          >
            Logout
          </Link>
        </span>
      </div>
    </div>
    <div
      className="sidenav-body"
    >
      <Link
        className="menu-item active"
        clickHandler={[Function]}
        id={null}
      >
        Dashboard
      </Link>
      <Link
        className="menu-item false"
        clickHandler={[Function]}
        id={null}
      >
        Meals
      </Link>
      <Link
        className="menu-item false"
        clickHandler={[Function]}
        id={null}
      >
        Menu
      </Link>
      <Link
        className="menu-item false"
        clickHandler={[Function]}
        id={null}
      >
        Orders
      </Link>
    </div>
  </div>
</div>
`;

exports[`SideNav renders correctly when meals tab is active 1`] = `
<div
  className="sidenav null"
  id="sidenav"
>
  <div
    className="sidenav-content"
  >
    <div
      className="sidenav-header"
    >
      <CloseIcon
        btnID={null}
        clickHandler={[Function]}
        divClass="d-none-md"
      />
      <div
        className="sidenav-title"
      >
        <h3>
          <Link
            href="/"
            replace={false}
            to="/"
          >
            BOOK-A-MEAL
          </Link>
        </h3>
        <div
          className="username-circle"
        >
          <p>
            F
          </p>
        </div>
        <p>
          Food Circle
        </p>
        <span>
          <Link
            className={null}
            clickHandler={[MockFunction]}
            id={null}
          >
            Logout
          </Link>
        </span>
      </div>
    </div>
    <div
      className="sidenav-body"
    >
      <Link
        className="menu-item false"
        clickHandler={[Function]}
        id={null}
      >
        Dashboard
      </Link>
      <Link
        className="menu-item active"
        clickHandler={[Function]}
        id={null}
      >
        Meals
      </Link>
      <Link
        className="menu-item false"
        clickHandler={[Function]}
        id={null}
      >
        Menu
      </Link>
      <Link
        className="menu-item false"
        clickHandler={[Function]}
        id={null}
      >
        Orders
      </Link>
    </div>
  </div>
</div>
`;

exports[`SideNav renders correctly when menu tab is active 1`] = `
<div
  className="sidenav null"
  id="sidenav"
>
  <div
    className="sidenav-content"
  >
    <div
      className="sidenav-header"
    >
      <CloseIcon
        btnID={null}
        clickHandler={[Function]}
        divClass="d-none-md"
      />
      <div
        className="sidenav-title"
      >
        <h3>
          <Link
            href="/"
            replace={false}
            to="/"
          >
            BOOK-A-MEAL
          </Link>
        </h3>
        <div
          className="username-circle"
        >
          <p>
            F
          </p>
        </div>
        <p>
          Food Circle
        </p>
        <span>
          <Link
            className={null}
            clickHandler={[MockFunction]}
            id={null}
          >
            Logout
          </Link>
        </span>
      </div>
    </div>
    <div
      className="sidenav-body"
    >
      <Link
        className="menu-item false"
        clickHandler={[Function]}
        id={null}
      >
        Dashboard
      </Link>
      <Link
        className="menu-item false"
        clickHandler={[Function]}
        id={null}
      >
        Meals
      </Link>
      <Link
        className="menu-item active"
        clickHandler={[Function]}
        id={null}
      >
        Menu
      </Link>
      <Link
        className="menu-item false"
        clickHandler={[Function]}
        id={null}
      >
        Orders
      </Link>
    </div>
  </div>
</div>
`;

exports[`SideNav renders correctly when orders tab is active 1`] = `
<div
  className="sidenav null"
  id="sidenav"
>
  <div
    className="sidenav-content"
  >
    <div
      className="sidenav-header"
    >
      <CloseIcon
        btnID={null}
        clickHandler={[Function]}
        divClass="d-none-md"
      />
      <div
        className="sidenav-title"
      >
        <h3>
          <Link
            href="/"
            replace={false}
            to="/"
          >
            BOOK-A-MEAL
          </Link>
        </h3>
        <div
          className="username-circle"
        >
          <p>
            F
          </p>
        </div>
        <p>
          Food Circle
        </p>
        <span>
          <Link
            className={null}
            clickHandler={[MockFunction]}
            id={null}
          >
            Logout
          </Link>
        </span>
      </div>
    </div>
    <div
      className="sidenav-body"
    >
      <Link
        className="menu-item false"
        clickHandler={[Function]}
        id={null}
      >
        Dashboard
      </Link>
      <Link
        className="menu-item false"
        clickHandler={[Function]}
        id={null}
      >
        Meals
      </Link>
      <Link
        className="menu-item false"
        clickHandler={[Function]}
        id={null}
      >
        Menu
      </Link>
      <Link
        className="menu-item active"
        clickHandler={[Function]}
        id={null}
      >
        Orders
      </Link>
    </div>
  </div>
</div>
`;
